psl-core/src/main/java/org/linqs/psl/application/inference/SGDStreamingInference.java

```diff
@@ -51,14 +51,8 @@ protected void completeInitialize() {
     }
 
     @Override
-    public void inference() {
-        log.info("Beginning inference.");
+    protected void internalInference() {
```

Comment:
Look like all (except lazy) `internalInference()` are now the same.
Instead, you can probably provide a default implementation of `internalInference()` that lazy overrides.